xformers/helpers/hierarchical_configs.py

```diff
@@ -19,15 +19,16 @@ class BasicLayerConfig:
     stride: int
     padding: int
     seq_len: int
+    feedforward: str
 
 
 def get_hierarchical_configuration(
-    layer_basic_configs: List[BasicLayerConfig],
+    layer_base_configs: List[BasicLayerConfig],
     layernorm_style: LayerNormStyle = LayerNormStyle.Pre,
     use_rotary_embeddings: bool = True,
     mlp_multiplier: int = 4,
-    dim_head=32,
-    feedforward="MLP",
```

Comment:
this implied that the same feedforward was being used for all the layers, which is not the case with efficientformer. Moving this setting to the per-layer base configuration, more flexible